src/axes/categoryAxis.ts

```diff
@@ -117,22 +120,101 @@ namespace Plottable.Axes {
-        stepWidth: downsampleRatio * scale.stepWidth(),
-      };
+    public tickMarkPosition(tickMarkPosition: ITickMarkPosition): this;
+    public tickMarkPosition(tickMarkPosition?: ITickMarkPosition): any {
+      if (arguments.length === 0) {
+        return this._tickMarkPosition;
+      } else {
+        this._tickMarkPosition = tickMarkPosition;
+      }
+      return this;
+    }
+
+    /**
+     * Gets the current alignment. If not set, null will be returned.
+     */
+    public tickTextAlignment(): string;
+    /**
+     * Set alignment of tick labels. Must be one of "left", "right",
+     * or "center", or a falsy value. Setting to a falsy value will
+     * reset tick label alignment to the default alignment.
+     *
+     * Note that alignment is applied before rotation (see "{@link
+     * tickLabelAngle}").
+     *
+     * @param {String} tickTextAlignment One of "left", "right", "center",
+     * or a falsy value.
+     *
+     * @returns {Category} The calling Category.
+     */
+    public tickTextAlignment(tickTextAlignment: string): this;
+    public tickTextAlignment(tickTextAlignment?: string): any {
+      if (arguments.length === 0) {
+        return this._tickTextAlignment;
+      }
+
+      if (! tickTextAlignment) {
+        this._tickTextAlignment = null;
+      } else {
+        let v = tickTextAlignment.toLowerCase();
+        if (v !== "left" && v !== "right" && v !== "center") {
+          throw new Error("tickTextAlignment '" + tickTextAlignment + "' not supported. Must be left, right, or center.");
+        }
+        this._tickTextAlignment = v;
+      }
+
+      return this;
+    }
+
+    /**
+     * Returns the current padding between tick labels and the
+     * opposite edge of the axis.
+     */
+    public tickTextPadding(): number;
+    /**
+     * Sets Padding between labels and outer edge of the axis (i.e.,
+     * the edge opposite the plot, as defined by the axis' {@link
+     * orientation}).
+     *
+     * Padding will only be applied when {@link tickTextAlignment} is
+     * set and either of the following conditions hold:
+     *
+     *   * Orientation is "left" or "right"; labels are not rotated;
+     *     for "left" orientation, {@link tickTextAlignment} must be
+     *     "left". For "right" orientation, {@link tickTextAlignment}
+     *     must be "right".
+     *
+     *   * Orientation is "top" or "bottom"; labels are rotated. For
+     *     "top" orientation and 90 rotation, alignment must "left";
+     *     for -90 rotation, alignment must be "right". For "bottom"
+     *     orientation and 90 rotation, alignment must "right"; For
+     *     -90 rotation, alignment must be "left".
+     *
+     * @param {Number} tickTextPadding Padding in pixels.
+     *
+     * @returns {Category} The calling Category.
+     */
+    public tickTextPadding(tickTextPadding: number): this;
+    public tickTextPadding(tickTextPadding?: number): any {
```

Comment:
I have a really hard time reading the comment for this API. What does the "opposite edge of the axis" refer to? I'm guessing that it will intelligently pad based on the orientation and alignment of the label and the position of the axis? I'm guessing its probably really intuitive once you see it in action, but the comment is long and confusing. Do you think it's necessary to list every case?